minecraft example fixes + VR controller support

shpowley opened the pull request:
- desktop mode + keyboard: general player movement related fixes
- VR mode + controllers: added VR controller support for player movement

bbohlender commented on examples/minecraft/src/VRPlayerControl.jsx:
```diff
@@ -0,0 +1,52 @@
+    a_button: null
+  }
+
+  useEffect(() => {
```
the evaluations made in this use effect better fit directly into the use frame imo, seems overlay complex

shpowley replied:
jump check placed directly in useFrame. useEffect removed.

bbohlender commented on examples/minecraft/src/VRPlayerControl.jsx:
```diff
@@ -0,0 +1,52 @@
+        backward: gamepad_info.thumbstick.yAxis > 0,
+        left: gamepad_info.thumbstick.xAxis < 0,
+        right: gamepad_info.thumbstick.xAxis > 0,
+        rotation: state.camera.rotation
```
this should probably something like `state.camera.getWorldRotation(helperEuler)`

shpowley replied:
implemented suggestion. see latest `VRPlayerControl.jsx`. was this done correctly?
no noticeable difference compared to state.camera.rotation ..VR headset turn has almost identical rotation on movement.

bbohlender commented:
@shpowley very nice PR
I have a few nitpicks, which I can also adress myself, but in general looks good to me
will merge, once I get the chance to try out myself :)

shpowley commented:
Implemented suggested fixes.
Added immersive VR right thumbstick player turning.
Also, added x-axis "wiggle tolerance" check to make forward/backward movement less "janky".
